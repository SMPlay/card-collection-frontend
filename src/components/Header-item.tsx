import React from "react";
import { createUseStyles, useTheme } from "react-jss";

const useStyles = createUseStyles({
  headerItem: {
    position: "relative",
    fontSize: 22,
    fontWeight: 600,
    maxWidth: 130,
    height: 27,
    padding: "0 10px",
    color: ({theme}) => theme.colorPrimary,
    background: "rgb(16, 118, 227)",
    transform: "skew(-20deg)",
    cursor: "pointer",
    border: "1px solid rgb(18, 18, 18)",
    "&::after": {
      position: "absolute",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      content: "''",
      width: "100%",
      height: "100%",
      opacity: 0,
      background: "linear-gradient(to top, rgba(255, 255, 255, .5), transparent)",
      transition: "opacity .2s ease",
    },
    "&:hover": {
      "&::after": {
        opacity: .6
      }
    }
  },
  headerItemChild: {
    position: "relative",
    zIndex: 100,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    transform: "skew(20deg)",
  }
});

export const HeaderItem: React.FC = ({ children }) => {
  const theme = useTheme();
  const styles = useStyles({theme});

  return (
    <div className={styles.headerItem}>
      <span className={styles.headerItemChild}>{children}</span>
    </div>
  );
}
