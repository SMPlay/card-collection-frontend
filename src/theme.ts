export const theme = {
  colorPrimary: "rgb(245, 81, 81)"
}
