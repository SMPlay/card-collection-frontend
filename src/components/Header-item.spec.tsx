import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "react-jss";

import { HeaderItem } from "./Header-item";
import { theme } from "../theme";

describe("Header item component", () => {
  it("should render component", () => {
    const tree = renderer.create(
     <ThemeProvider theme={theme}>
        <HeaderItem></HeaderItem>
     </ThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
