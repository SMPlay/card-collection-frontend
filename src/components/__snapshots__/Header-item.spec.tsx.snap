// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header item component should render component 1`] = `
<div
  className="headerItem-0-2-1 headerItem-d0-0-2-3"
>
  <span
    className="headerItemChild-0-2-2"
  />
</div>
`;
